### PR TITLE
Features/bootstrap

### PR DESCRIPTION
**Pull request by AGloaguen**
Begining of rgcca_predict by Fabien. Handle new bootstrap resampling strategy also.

### CODE REVIEW

File: R/set_rgcca.R [hunk 1]
```diff
@@ -1,141 +1,80 @@
...
-    if(is.null(NA_method)){    NA_method <- rgcca_res$call$NA_method }
-    if(is.null(scheme)){     scheme <- rgcca_res$call$scheme}
-    if(is.null(bias)){     bias <- rgcca_res$call$bias}
-    if(is.null(method)){   method <- rgcca_res$call$method}
-    if(is.null(init)){     init <- rgcca_res$call$init}
-    if(is.null(ncomp)){        ncomp <- rgcca_res$call$ncomp}
-     if (is.null(blocks)) {
-      #  blocks <- rgcca_res$call$blocks
-         blocks = rgcca_res$call$raw
-         blocks = descale(blocks)
```
Comment (AGloaguen) [inline R/set_rgcca.R:36]: I might suppress this part as now blocks are raw, so no reason to `descale`

File: R/set_rgcca.R [hunk 2]
```diff
@@ -1,141 +1,80 @@
...
-
-    } else {
-
-        if (!is.null(blocks) && !missing(sparsity) && missing(tau))
-            stop_rgcca(paste0("tau parameter required for ", tolower(method), "instead of sparsity."))
-
-        par <- "tau"
-        penalty <- tau
-    }
-
```
Comment (AGloaguen) [inline R/set_rgcca.R:77]: This check on `tolower(method)` is already performed in `rgcca`. In order for us to be sure we can remove it, I decided to suppress `method` from the input parameters of `set_rgcca`.

File: R/set_rgcca.R [hunk 3]
```diff
@@ -1,141 +1,80 @@
-# inds : individuals removed from the blocks (for crossvalidation)
-# affects same parameters as rgcca_res (or other ones if specified) on a subset 
-# of individuals determined by inds (individuals to remove)
-set_rgcca <- function(
-    rgcca_res,
-    blocks = NULL,
-    connection = NULL,
-    tau = 1,
-    sparsity = 1,
```
Comment (AGloaguen) [inline R/set_rgcca.R:9]: I do not understand why `sparsity` and `tau` are by default set to `1`. Previously, is `block` was `NULL`, this default value was used instead of the ones in `rgcca_res$call`. I decided to set them by default as `NULL` as for the orther paramters and to check if they are `NULL`. If this is the case, I use their value in `rgcca_res$call`.

**Review by AGloaguen — COMMENTED** [review]
On the comparison between paralellization on a list of indexes or on a list of already sampled blocks :

image

In any case, with the new strategy that only checks the standard deviation accross all bootstrap samples, the choice is made to paralellize on a list of index because there is no need anymore to sample all the blocks.

N.B. : this test was performed on the Glioma dataset

**Review by AGloaguen — COMMENTED** [review]
One final question, I decided not to add a test in `bootstrap` for variable with null variance because it is based on a `rgcca` object that was previously tested for that. However, if a user decide by himself to add a null variance variable inside `rgcca_res$call$raw` and then run `bootsrap(rgcca_res, n_boot = 1)`, `generate_resampling` will generate an error that is not handled. Do you agree with that?
